designer/client/src/components/graph/node-modal/editors/expression/MockedOutputField.tsx

```diff
@@ -0,0 +1,101 @@
+import { EditorType } from "./types";
+
+const MOCKED_OUTPUT_IN_NODE_FIELD_NAME = "mockedOutput";
+const MOCK_EXPRESSION_PARAMETER_NAME = "mockExpression";
+const EXPRESSION_TEXT_PATH = `${MOCKED_OUTPUT_IN_NODE_FIELD_NAME}.${MOCK_EXPRESSION_PARAMETER_NAME}.expression`;
+
+const MOCK_EXPRESSION_HINT_TEXT =
+    "If provided the expression will be evaluated in tests instead of invoking real enricher behaviour (e.g. instead of calling a service/database).\n\n" +
+    "Evaluated value will be placed under output variable name during tests.";
+
+const UnknownTypingResult = {
+    params: [],
+    type: "Unknown",
+    display: "Unknown",
+    refClazzName: "java.lang.Object",
+};
+
+// we construct artificial and hardcoded parameter definition to reuse EditableEditor for mockExpression field
+const MockExpressionParameter: UIParameter = {
+    additionalVariables: {},
+    branchParam: false,
+    defaultValue: { expression: "", language: "spel" },
+    editor: { type: EditorType.SPEL_PARAMETER_EDITOR },
+    label: "",
+    name: MOCK_EXPRESSION_PARAMETER_NAME,
+    typ: UnknownTypingResult,
+    variablesToHide: [],
+};
+
+type Props = {
+    editedNode: NodeType;
+    isEditMode: boolean;
+    showValidation: boolean;
+    showSwitch: boolean;
+    findAvailableVariables: ReturnType<typeof ProcessUtils.findAvailableVariables>;
+    setNodeDataAt: <T>(propToMutate: string, newValue: T, defaultValue?: T) => void;
+    errors: NodeValidationError[];
+};
+
+function MockedOutputField(props: Props): JSX.Element {
+    const { editedNode, isEditMode, showValidation, showSwitch, findAvailableVariables, setNodeDataAt, errors } = props;
+    const [mockExpression, setMockExpression] = useState(() => {
+        return get(editedNode, MOCKED_OUTPUT_IN_NODE_FIELD_NAME)?.mockExpression || { expression: "", language: "spel" };
+    });
+    const [isMarked] = useDiffMark();
+    const { t } = useTranslation();
+    const readOnly = !isEditMode;
+
+    const onValueChange: OnValueChange = useCallback(
+        (value: ExpressionObj) => {
+            setMockExpression(value);
+            if (value.expression.length > 0) {
+                setNodeDataAt(MOCKED_OUTPUT_IN_NODE_FIELD_NAME, { type: "SingleMockExpression", mockExpression: value });
+            } else {
+                setNodeDataAt(MOCKED_OUTPUT_IN_NODE_FIELD_NAME, null);
+            }
+        },
+        [setNodeDataAt],
+    );
+
+    const renderMockExpressionParameterLabel = (): JSX.Element => {
+        return <FieldLabel label={t("nodes.enricher.mockExpression", "Mock")} hintText={MOCK_EXPRESSION_HINT_TEXT} />;
+    };
+
+    const variableTypes = useMemo(() => findAvailableVariables(editedNode.id), [findAvailableVariables, editedNode.id]);
+    const testResultsState = useTestResults();
+
+    return (
+        <ExpressionTestResults fieldName={MOCK_EXPRESSION_PARAMETER_NAME} resultsToShow={testResultsState.testResultsToShow}>
+            <EditableEditor
```

Comment:
Currently, we are in the middle in the migration to `Json template | Expression` combination of editors in almost all places. So far, we did it conservatively, only in places where at the end, json is processed somehow, but there is the plan to use this approach almost everywhere. Let's use this combination here as well.

Reply:
fixed. 
With jsontemplate there was as a few issues:
* it didn't allowed null values if expected typing wasn't directly set to `TypedNull` - I fixed it to allow null always despite of typing.
* it jsontempate is not checked against expected typing. It seems that in actual usages there was always `Unknown/Json` type set as expected type (json schemaless kafka source, http enricher body, event generator). I added additional post compile type check for mock expression. We can discuss whether it should be moved to JsonTemplateParser.

Reply:
Why not to move this validation to json template parser now? It would be consistent with spel parser.